src/formatDate.js

```diff
@@ -8,7 +8,44 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const slicedDate = date.split(fromFormat[3]);
+  const newSeparator = toFormat[3];
+
+  const dayIndex = fromFormat.indexOf('DD');
+  const monthIndex = fromFormat.indexOf('MM');
+  const yearIndex = Math.abs(dayIndex + monthIndex - 3);
+
+  const newDayIndex = toFormat.indexOf('DD');
+  const newMonthIndex = toFormat.indexOf('MM');
+  const newYearIndex = Math.abs(newDayIndex + newMonthIndex - 3);
+
+  let formatedYear = slicedDate[yearIndex];
+  const newDateParts = [];
+
+  if (
+    fromFormat[yearIndex].length === 4 &&
+    toFormat[newYearIndex].length === 2
+  ) {
+    formatedYear = slicedDate[yearIndex].slice(2);
```

Comment:
It's good that you are slicing the last two digits for the year when converting from 'YYYY' to 'YY'. However, you could use 'substr' instead of 'slice' for better semantic meaning as you are not really slicing out a middle part but getting a substring from the end.